### PR TITLE
feat(STADTPULS-248): sensor page pagination

### PR DESCRIPTION
**Pull request by dnsos**
This PR adds pagination to the sensors overview page. 

We are displaying 15 sensor's per page, which is more than what we currently have in our staging environment. So, in order to properly assess the pagination UI, you may want to reduce `MAX_SENSORS_PER_PAGE` in `pages/sensors/index.tsx` in your local setup; or simply check the Storybook entry.

### CODE REVIEW

**Review by vogelino — COMMENTED** [review]
Great Work! Nice straightforward UI/UX too! The code is clean and tested, however, I left a few comments. Also, I noticed that when you open the URL with a page number that doesn't exist, an empty page with "Keine Sensoren Vorhanden" is  displayed (see screenshot):

Screen Shot 2021-11-10 at 10 06 27 AM

Shouldn't we rather default to the first page? What do you think?

File: pages/sensors/index.tsx [hunk 1]
```diff
@@ -1,43 +1,92 @@
...
     );
+
+  const pageCount = Math.ceil(sensorsCount / MAX_SENSORS_PER_PAGE);
+
   return (
     <div className='container mx-auto max-w-8xl py-24 px-4'>
-      <h1
-        className={[
-          "font-bold text-xl sm:text-2xl md:text-3xl font-headline",
+      <hgroup
```
[the remark this answers is not in the document]
Comment (dnsos) [reply pages/sensors/index.tsx]: Interesting, I didn't know this. Removed the `hgroup` with the latest commits!

File: pages/sensors/index.tsx [hunk 2]
```diff
@@ -1,43 +1,92 @@
...
+import { getPublicSensors } from "@lib/requests/getPublicSensors";
 
 interface SensorsOverviewPropType {
   sensors: ParsedSensorType[];
+  sensorsCount: number;
+  page: number;
 }
 
-export const getServerSideProps: GetServerSideProps = async () => {
+const MAX_SENSORS_PER_PAGE = 15;
```
[the remark this answers is not in the document]
Comment (dnsos) [reply pages/sensors/index.tsx:17]: You're right, and it was an issue of my code.

I hadn't treated the `0` (zero) as a valid `rangeStart` which resulted in `getPublicSensors` falling back to fetching _all_ sensors. These were then shown on the first page. The following pages' `rangeStart`'s are obviously all higher than zero. That's why it was respected for them.

Fixed this in 2b71c81.

**Comment by dnsos** [comment]
> Shouldn't we rather default to the first page? What do you think?

Good idea, thanks for pointing to this. I'm gonna change that.

**Review by vogelino — APPROVED** [review]
Great corrections! Thx!

**Comment by dnsos** [comment]
@vogelino, thanks for the feedback. Can you take one more look at the latest commit? It ensures that we default to the first page if someone queries a `sensorPage` that exceeds our page count.

**Review by vogelino — APPROVED** [review]
👍

**Pull request merged**